### PR TITLE
Add docker archive scan

### PR DESCRIPTION
**Pull request by Jguer**
- Add `ggshield scan docker` command

### CODE REVIEW

**Review by julienc91 — CHANGES_REQUESTED** [review]

**Review by Walz — COMMENTED** [review]
I've tested the code on ~300 images and except the issue with `.decode(errors="replace").replace("\0", "�")`, everything looks good to me.

File: ggshield/scan/docker.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+        file = layer_archive.extractfile(file_info)
+        if file is None:
+            continue
+
+        file_content_raw = file.read()
+        if len(file_content_raw) > MAX_FILE_SIZE * 0.95:
+            continue
+
+        file_content = file_content_raw.decode(errors="replace").replace("\0", "�")
```
[the remark this answers is not in the document]
Comment (Jguer) [reply ggshield/scan/docker.py:162]: :sweat_smile:  not sure why actually, didn't write this part but I'll be changing it

File: ggshield/scan/docker.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+        if any(file_info.path.endswith(ext) for ext in DEFAULT_EXTENSION_BANLIST):
+            continue
+
+        file = layer_archive.extractfile(file_info)
+        if file is None:
+            continue
+
+        file_content_raw = file.read()
+        if len(file_content_raw) > MAX_FILE_SIZE * 0.95:
```
[the remark this answers is not in the document]
Comment (Jguer) [reply ggshield/scan/docker.py:159]: really going to the tightest of limits results in weird edge cases, i'd leave this as a safety margin

File: ggshield/scan/docker.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    )
+
+
+def _should_scan_layer(layer_info: Dict) -> bool:
+    """
+    Returns True if a layer should be scanned, False otherwise.
+    Only COPY and ADD layers should be scanned.
+    """
+    cmd = layer_info["created_by"]
+    return "COPY" in cmd or "ADD" in cmd
```
[the remark this answers is not in the document]
Comment (Jguer) [reply ggshield/scan/docker.py:118]: it was perfect, it just missed r"\b(copy|add)\b" to work

File: README.md [hunk 4]
```diff
@@ -200,13 +200,13 @@ Commands:
...
-```
-Usage: ggshield scan docker [OPTIONS] IMAGE_NAME
+  ```
+  Usage: ggshield scan docker [OPTIONS] IMAGE_NAME
 
-  ggshield will try to pull the image if it's not available on the local
-Options:
-  -h, --help  Show this message and exit.
-```
+    ggshield will try to pull the image if it's not available localy
```
Comment (julienc91) [inline README.md:206]: `locally`

**Review by julienc91 — APPROVED** [review]

**Review by Walz — APPROVED** [review]

**Pull request merged**